File: src/main/java/jukebot/commands/Select.java
package jukebot.commands;

import jukebot.JukeBot;
import jukebot.audioutilities.AudioHandler;
import jukebot.audioutilities.SongResultHandler;
import jukebot.utils.Command;
import jukebot.utils.CommandProperties;
import jukebot.utils.Context;

@CommandProperties(description = "Search YouTube and select from up to 5 tracks", aliases = {"search", "sel", "s"}, category = CommandProperties.category.CONTROLS)
public class Select implements Command {

    public void execute(final Context context) {

        if (context.getArgString().isEmpty()) {
            context.sendEmbed("YouTube Search", "Specify what to search for.");
            return;
        }

        final AudioHandler player = context.getAudioPlayer();
        final Boolean voiceConnected = context.ensureVoice();

        if (!voiceConnected) {
            return;
        }

        if (!player.isPlaying()) {
            player.setChannel(context.getChannel().getIdLong());
        }

        JukeBot.playerManager.loadItem("ytsearch:" + context.getArgString(), new SongResultHandler(context, player, true));
    }
}

File: src/main/java/jukebot/commands/Select.kt
package jukebot.commands

import jukebot.JukeBot
import jukebot.framework.Command
import jukebot.framework.CommandCategory
import jukebot.framework.CommandProperties
import jukebot.framework.Context

@CommandProperties(
    description = "Search YouTube and select from up to 5 tracks",
    aliases = ["search", "sel", "s", "find", "add"],
    category = CommandCategory.PLAYBACK
)
class Select : Command(ExecutionType.TRIGGER_CONNECT) {

    override fun execute(context: Context) {
        val player = context.getAudioPlayer()

        if (!player.isPlaying) {
            player.channelId = context.channel.idLong
        }

        JukeBot.playerManager.loadIdentifier("ytsearch:${context.argString}", context, player, true)
    }
}
